core/net/src/service.rs

```diff
@@ -19,27 +23,44 @@ pub fn central_net_addr() -> std::io::Result<SocketAddr> {
         .expect("central net hub addr needed"))
 }
 
+#[inline]
+fn net_node_id(node_id: &NodeId) -> String {
+    format!("{}/{:?}", net::BUS_ID, node_id)
+}
+
+fn parse_from_addr(from_addr: &str) -> anyhow::Result<(NodeId, NodeId, &str)> {
```

Comment:
Could de done easier with a regex: `/from/(\w+)/to/(\w+)(/[\w/]+)`

Reply:
it won't:

- regex compilation runs in runtime. so i need lazy static or something.

does not looks very simpler?. and it is impossible to return borrowed slice from an argument. (or i do not know how to do this). so it has additional allocations on heap.

```rust
fn parse_from_addr(from_addr: &str) -> anyhow::Result<(NodeId, NodeId, String)> {
    let re = regex::Regex::new("/from/(\\w+)/to/(\\w+)(/[\\w/]+)")?;
    if let Some(c) = re.captures(from_addr) {
        let (from_node_id, to_node_id, service_id) = (&c[1], &c[2], &c[3]);
        return Ok((from_node_id.parse()?, to_node_id.parse()?, service_id.to_owned()));
    }
    Err(anyhow!("invalid net-from destination: {}", from_addr))
}
```

Reply:
It's clearer to see what the function does, but if it's too much trouble then never mind.